core/src/executor/select/mod.rs

```diff
@@ -74,24 +76,29 @@ async fn fetch_blended<'a>(
         let row = Some(row);
         let columns = Rc::clone(&columns);
 
-        Ok(BlendContext::new(table.get_alias(), columns, row, None))
+        Ok(BlendContext::new(relation.get_alias(), columns, row, None))
     });
-
     Ok(stream::iter(rows))
 }
 
-fn get_labels<'a>(
+pub fn get_labels<'a>(
     projection: &[SelectItem],
     table_alias: &str,
     columns: &'a [String],
-    join_columns: &'a [(&String, Vec<String>)],
+    join_columns: Option<&'a [(&String, Vec<String>)]>,
```

Comment:
From now, we should be able to get labels before join so that join_columns become Option